File: android_demo_project/app/src/main/java/com/sickworm/jugg/demo/testcase/databinding/DemoBindingAdapters.java
package com.sickworm.jugg.demo.testcase.databinding;

import android.view.View;

import androidx.databinding.BindingAdapter;

/**
 * Provides custom DataBinding adapters used by incremental compilation cases.
 */
public final class DemoBindingAdapters {

    private DemoBindingAdapters() {
    }

    @BindingAdapter("android:visibility")
    public static void setBooleanVisibility(View view, boolean visible) {
        view.setVisibility(visible ? View.VISIBLE : View.GONE);
    }
}

File: android_demo_project/app/src/main/java/com/sickworm/jugg/demo/testcase/databinding/DemoBindingAdapters.kt
package com.sickworm.jugg.demo.testcase.databinding

import android.view.View
import androidx.databinding.BindingAdapter

/**
 * Provides custom DataBinding adapters used by incremental compilation cases.
 */
object DemoBindingAdapters {

    @JvmStatic
    @BindingAdapter("android:visibility")
    fun setBooleanVisibility(view: View, visible: Boolean) {
        view.visibility = if (visible) View.VISIBLE else View.GONE
    }
}
